sentry/src/main/java/io/sentry/DateUtils.java

```diff
@@ -3,96 +3,72 @@
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
 /** Utilities to deal with dates */
 @ApiStatus.Internal
 public final class DateUtils {
   private static final String UTC = "UTC";
+  // ISO 8601
   private static final String ISO_FORMAT = "yyyy-MM-dd'T'HH:mm:ss'Z'";
   private static final String ISO_FORMAT_WITH_MILLIS = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
-  private static final ThreadLocal<SimpleDateFormat> SDF_ISO_FORMAT_WITH_MILLIS_UTC =
+
+  // if UTC is not found, it fallback to "GMT" which is UTC equivalent
+  private static final @NotNull TimeZone UTC_TIMEZONE = TimeZone.getTimeZone(UTC);
+
+  private static final @NotNull ThreadLocal<SimpleDateFormat> SDF_ISO_FORMAT_WITH_MILLIS_UTC =
       new ThreadLocal<SimpleDateFormat>() {
         @Override
         protected SimpleDateFormat initialValue() {
-          final TimeZone tz = TimeZone.getTimeZone(UTC);
           final SimpleDateFormat simpleDateFormat =
               new SimpleDateFormat(ISO_FORMAT_WITH_MILLIS, Locale.ROOT);
-          simpleDateFormat.setTimeZone(tz);
+          simpleDateFormat.setTimeZone(UTC_TIMEZONE);
           return simpleDateFormat;
         }
       };
-  private static final ThreadLocal<SimpleDateFormat> SDF_ISO_FORMAT_WITH_MILLIS =
-      new ThreadLocal<SimpleDateFormat>() {
-        @Override
-        protected SimpleDateFormat initialValue() {
-          return new SimpleDateFormat(ISO_FORMAT_WITH_MILLIS, Locale.ROOT);
-        }
-      };
-  private static final ThreadLocal<SimpleDateFormat> SDF_ISO_FORMAT =
+
+  private static final @NotNull ThreadLocal<SimpleDateFormat> SDF_ISO_FORMAT_UTC =
       new ThreadLocal<SimpleDateFormat>() {
         @Override
         protected SimpleDateFormat initialValue() {
-          return new SimpleDateFormat(ISO_FORMAT, Locale.ROOT);
+          final SimpleDateFormat simpleDateFormat = new SimpleDateFormat(ISO_FORMAT, Locale.ROOT);
+          simpleDateFormat.setTimeZone(UTC_TIMEZONE);
+          return simpleDateFormat;
         }
       };
 
   private DateUtils() {}
 
   /**
-   * Get date formatted as expected by Sentry.
-   *
-   * @param date the current date with local timezone
-   * @return the ISO formatted UTC date with millis precision.
-   */
-  public static @NotNull String getTimestampIsoFormat(final @NotNull Date date) {
-    return SDF_ISO_FORMAT_WITH_MILLIS_UTC.get().format(date);
-  }
-
-  /**
-   * Get the current date and time as ISO UTC
+   * Get the current Date (UTC)
    *
-   * @return the ISO UTC date and time
+   * @return the UTC Date
    */
   @SuppressWarnings("JdkObsolete")
-  public static @NotNull Date getCurrentDateTime() throws IllegalArgumentException {
-    final String timestampIsoFormat = getTimestampIsoFormat(new Date());
-    return getDateTime(timestampIsoFormat);
+  public static @NotNull Date getCurrentDateTime() {
+    final Calendar calendar = Calendar.getInstance(UTC_TIMEZONE);
```

Comment:
`m`: Does it maybe make sense to move this to `private static final Calendar calendar = Calendar.getInstance(UTC_TIMEZONE);` so we save creating an instance on each call?

Reply:
no, Calendar is equivalent to `new Date()`, we need to create a new one to get the current date/time

Reply:
the thing is, even that calendat.getTime() returns a new Date, makes sense that we'd cache it, but when you look at the Calendar class, it has some Global fields like `isTimeSet` etc being used to generate new Times, so I'm afraid that this will give us wrong values, as I don't know much about the Calendar class, I've used to work with Jodatime or other tools.
Do you know if its safe to Cache the Calendar and use only it?

Reply:
`Calendar` is not thread-safe so adding to a static field won't work.